cmd/convertor/builder/builder_utils.go

```diff
@@ -22,52 +22,99 @@ import (
 	"context"
 	"crypto/sha256"
 	"encoding/json"
+	"fmt"
 	"io"
 	"os"
 	"path"
 
-	"github.com/containerd/accelerated-container-image/pkg/snapshot"
 	"github.com/containerd/containerd/archive/compression"
 	"github.com/containerd/containerd/content"
 	"github.com/containerd/containerd/errdefs"
+	"github.com/containerd/containerd/images"
+	"github.com/containerd/containerd/log"
+	"github.com/containerd/containerd/platforms"
 	"github.com/containerd/containerd/remotes"
 	"github.com/containerd/continuity"
 	"github.com/opencontainers/go-digest"
 	specs "github.com/opencontainers/image-spec/specs-go/v1"
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
+
+	"github.com/containerd/accelerated-container-image/pkg/snapshot"
 )
 
-func fetchManifestAndConfig(ctx context.Context, fetcher remotes.Fetcher, manifestDesc specs.Descriptor) (specs.Manifest, specs.Image, error) {
-	// get manifest
-	rc, err := fetcher.Fetch(ctx, manifestDesc)
+func fetch(ctx context.Context, fetcher remotes.Fetcher, desc specs.Descriptor, target any) error {
```

Comment:
Nit: Should we use the `any` type? This requires go1.18+ but our [QUICKSTART](https://github.com/containerd/accelerated-container-image/blob/main/docs/QUICKSTART.md#compile-from-source) says that only go1.16+ is needed. Should we rewrite `interface{}` to `any` and change QUICKSTART now? Or we could do this in the future？ @BigVan cc

Reply:
@WaberZhuang  we should update doc. we are always trying to keep the golang version consistent with the containerd.

Reply:
OK, and I recommend always using `any` instead of `interface{}` in the future.